compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirDeclarationsResolveTransformer.kt

```diff
@@ -509,6 +511,44 @@ open class FirDeclarationsResolveTransformer(transformer: FirBodyResolveTransfor
         return result.compose()
     }
 
+    /*
+     * Suppose a function without an explicit return type just returns an anonymous object:
+     *
+     *   fun foo(...) = object : ObjectSuperType {
+     *     override fun ...
+     *   }
+     *
+     * Without unwrapping, the return type ended up with that anonymous object (<no name provided>), while the resolved super type, which
+     * acts like an implementing interface, is a better fit.
+     */
+    private fun FirTypeRef.unwrapAnonymousObjectSupertypeIfNeeded(): FirTypeRef {
+        val anonymousObject =
+            ((((this as? FirResolvedTypeRef)
+                ?.type as? ConeClassLikeType)
+                ?.lookupTag as? ConeClassLookupTagWithFixedSymbol)
+                ?.symbol as? FirAnonymousObjectSymbol)?.fir
+                ?: return this
+        // TODO: replace this ad-hoc condition with a better handling
+        //   Conflict: diagnostic test properties.inferenceFromGetters.objectExpression
+        //     has a property whose getter returns an anonymous object with own members, which are later invoked.
+        //     If we unwrap the type here, those member accesses become unresolved calls.
```

Comment:
Discussed with other folks (@madsager @pyos), probably the current result---resolving a member in an anonymous object---is not a desired behavior, setting the right return type here, which results in unresolved calls there, would be correct?

Reply:
I just learned when to _not_ expose anonymous object types. Now that remaining bb test is working; also, some `invisible` members are correctly hidden/unresolved.

Reply:
Btw, special thanks to @pyos !